agent/taskresource/volume/volume.go

```diff
@@ -98,11 +104,40 @@ func (vol *VolumeResource) GetCreatedAt() time.Time {
 	return vol.createdAtUnsafe
 }
 
+// Create performs resource creation
+func (vol *VolumeResource) Create() error {
+	volumeResponse := vol.client.CreateVolume(
+		vol.Name,
+		vol.Driver,
+		vol.DriverOpts,
+		vol.Labels,
+		engine.CreateVolumeTimeout)
+
+	if volumeResponse.Error != nil {
+		return volumeResponse.Error
+	}
+
+	// set readonly field after creation
+	vol.Mountpoint = volumeResponse.DockerVolume.Mountpoint
+	return nil
+}
+
+// Cleanup performs resource cleanup
+func  (vol *VolumeResource) Cleanup() error {
+	err := vol.client.RemoveVolume(vol.Name, engine.RemoveVolumeTimeout)
+
+	if err != nil {
+		return err
```

Comment:
Is this error is handled or logged somewhere?

Reply:
see previous comment by aaithal@, I removed it to log it from caller, where more contextual information (like task id) would be available.